frame/multi-account/src/lib.rs

```diff
@@ -0,0 +1,1506 @@
+
+			let multi_account_index = MultiAccountIndex::get()
+				.checked_add(1)
+				.expect("multi account indices will never reach 2^64 before the death of the universe; qed");
+
+			let id = Self::multi_account_id(multi_account_index);
+
+			let deposit = T::MultiAccountDepositBase::get()
+				+ T::MultiAccountDepositFactor::get() * (other_signatories.len() as u32 + 1).into();
+
+			MultiAccountIndex::put(multi_account_index);
+
+			T::Currency::reserve(&who, deposit)?;
+
+			<MultiAccounts<T>>::insert(&id, MultiAccountData {
+				threshold,
+				signatories,
+				deposit,
+				depositor: who.clone(),
+			});
+
+			Self::deposit_event(RawEvent::NewMultiAccount(who, id));
+
+			Ok(())
+		}
+
+		/// Update an existing multi account with a new threshold and new signatories.
+		///
+		/// This maintains the multi account's account id.
+		///
+		/// Payment: `MultiAccountDepositBase` will be reserved, plus the number of signatories times
+		/// `MultiAccountDepositFactor`. It is returned when the multi account is updated or removed. This call will
+		/// unreserve the previous deposit.
+		///
+		/// The dispatch origin for this call must be _Signed_ and be equal to the multi account ID. It can be
+		/// dispatched by a `call` from this module.
+		///
+		/// Any removed signatories that have approved multisig transactions will no longer be counted as valid
+		/// approvals. Removing a signatory that has initiated/is the owner of a multisig transaction will not
+		/// cancel the multisig transaction, but the owner can always cancel the multisig transaction, even after
+		/// leaving the signatories. Such a transaction can still be dispatched, as long as the threshold
+		/// is met by the current signatories.
+		///
+		/// - `threshold`: The total number of approvals dispatches from this multi account require before they are
+		///    executed.
+		/// - `signatories`: The accounts who can approve dispatches from this multi account. May not be empty.
+		///
+		/// # <weight>
+		/// - `O(S)`.
+		/// - One balance-reserve and unreserve operation.
+		/// - One insert with `O(S)` where `S` is the number of
+		///   signatories. `S` is capped by `MaxSignatories`, with weight being proportional.
+		/// - One encode & hash, both of complexity `O(S)`.
+		/// - I/O: 1 read `O(S)`, 1 mutate `O(S)`.
+		/// - One event.
+		/// - Storage: inserts one item, value size bounded by `MaxSignatories`, with a
+		///   deposit taken for its lifetime of
+		///   `MultiAccountDepositBase + signatories.len() * MultiAccountDepositFactor`.
+		/// # </weight>
+		#[weight = <SigsLen<T::AccountId>>::new()]
+		fn update(origin,
+			threshold: u16,
+			signatories: Vec<T::AccountId>
+		) -> DispatchResult {
+			let who = ensure_signed(origin)?;
+			ensure!(threshold >= 1, Error::<T>::ZeroThreshold);
+			let max_sigs = T::MaxSignatories::get() as usize;
+			ensure!(signatories.len() >= 2, Error::<T>::TooFewSignatories);
```

Comment:
Why must it be `>= 2`?

Reply:
The `create` function requires at least 1 other signatory, meaning each multi account requires 2 or more sigantories. Likewise, we require an update to set at least 2 signatories. This is in line with `as_multi` in the `utility` pallet. 

I agree though that we would allow more cases if we allow 1 signatory (proxy accounts that can be extendend later etc...). What do you think?